<?xml version="1.0"?>
<robot xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor"
       xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller"
       xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface">

  <property name="M_PI" value="3.1415926535897931" />
 

  <macro name="m3_neck_transmission">
  <transmission name="head_t0" type="pr2_mechanism_model/SimpleTransmission">
      <actuator name="head_a0"/>
      <joint name="head_j0"/>
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>

  <transmission name="head_t1" type="pr2_mechanism_model/SimpleTransmission">
      <actuator name="head_a1"/>
      <joint name="head_j1"/>
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>
     
  <transmission name="head_t2" type="pr2_mechanism_model/SimpleTransmission">
      <actuator name="head_a2"/>
      <joint name="head_j2"/>
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>
  
  <transmission name="head_t3" type="pr2_mechanism_model/SimpleTransmission">
      <actuator name="head_a3"/>
      <joint name="head_j3"/>
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>

</macro>

  <macro name="m3_eye_transmission">
  <transmission name="head_t4" type="pr2_mechanism_model/SimpleTransmission">
      <actuator name="head_a4"/>
      <joint name="head_j4"/>
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>
  
  <transmission name="head_t5" type="pr2_mechanism_model/SimpleTransmission">
      <actuator name="head_a5"/>
      <joint name="head_j5"/>
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>
  
  <transmission name="head_t6" type="pr2_mechanism_model/SimpleTransmission">
      <actuator name="head_a6"/>
      <joint name="head_j6"/>
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>
  <!-- Temp for gazebo -->
  <transmission name="head_t7" type="pr2_mechanism_model/SimpleTransmission">
      <actuator name="head_a7"/>
      <joint name="head_j7"/>
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>
  
  <!--transmission name="head_t7_rt_eyelid_top" type="pr2_mechanism_model/SimpleTransmission">
      <actuator name="head_a7_rt_eyelid_top"/>
      <joint name="head_j7_rt_eyelid_top"/>
    <mechanicalReduction>1</mechanicalReduction>
  </transmission-->

  <transmission name="head_t7_rt_eyelid_bottom" type="pr2_mechanism_model/SimpleTransmission">
      <actuator name="head_a7_rt_eyelid_bottom"/>
      <joint name="head_j7_rt_eyelid_bottom"/>
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>
  
  <transmission name="head_t7_lt_eyelid_top" type="pr2_mechanism_model/SimpleTransmission">
      <actuator name="head_a7_lt_eyelid_top"/>
      <joint name="head_j7_lt_eyelid_top"/>
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>

  <transmission name="head_t7_lt_eyelid_bottom" type="pr2_mechanism_model/SimpleTransmission">
      <actuator name="head_a7_lt_eyelid_bottom"/>
      <joint name="head_j7_lt_eyelid_bottom"/>
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>
</macro>
  
  <macro name="m3_ear_transmission">
  <transmission name="head_t8" type="pr2_mechanism_model/SimpleTransmission">
      <actuator name="head_a8"/>
      <joint name="head_j8"/>
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>

  <transmission name="head_t9" type="pr2_mechanism_model/SimpleTransmission">
      <actuator name="head_a9"/>
      <joint name="head_j9"/>
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>

  <transmission name="head_t10" type="pr2_mechanism_model/SimpleTransmission">
      <actuator name="head_a10"/>
      <joint name="head_j10"/>
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>

  <transmission name="head_t11" type="pr2_mechanism_model/SimpleTransmission">
      <actuator name="head_a11"/>
      <joint name="head_j11"/>
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>
</macro>

  <macro name="m3_head_transmission">
    <m3_neck_transmission/>
    <m3_eye_transmission/>
    <m3_ear_transmission/>
  </macro>


</robot>
